Fix/hdr metadata parsing

vansante opened the pull request:
See: https://github.com/vansante/go-ffprobe/issues/55

shanehughes1990 committed:
fix: handle HDR metadata fields as flexible int/string types

HDR metadata fields (red_x, red_y, green_x, green_y, blue_x, blue_y,
white_point_x, white_point_y, min_luminance, max_luminance) can be
returned by ffprobe as either integers or fractional strings like
'34000/50000'.

Introduced FlexInt type that handles both formats via custom JSON
unmarshaling, preventing 'cannot unmarshal string into Go struct
field' errors when processing HDR video content.

shanehughes1990 committed:
fix(sidedata)!: use float64 for HDR metadata to preserve decimal precision

BREAKING CHANGE: Renamed FlexInt to FlexFloat (int → float64)

Fixes precision loss in HDR mastering display metadata where fractional
values like "34000/50000" (0.68) were truncated to 0. Changed underlying
type from int to float64 and removed int() truncation in fraction parsing.

- Achieved 100% test coverage for sidedata.go and ffprobe.go runProbe
- Added 90+ test cases for all side data types and error paths
- Addresses issue #55 concern about precision loss

Refs: #55

shanehughes1990 committed:
Chopping HDR test file to 1 second for ~6MB filesize.

shanehughes1990 commented:
--- PASS: Test_ProbeAllAssets (0.59s)
    --- PASS: Test_ProbeAllAssets/test.avi (0.39s)
    --- PASS: Test_ProbeAllAssets/test.mov (0.07s)
    --- PASS: Test_ProbeAllAssets/test.mp4 (0.06s)
    --- PASS: Test_ProbeAllAssets/test.ts (0.08s)
PASS
coverage: 30.9% of statements
ok      gopkg.in/vansante/go-ffprobe.v2 0.801s  coverage: 30.9% of statements

Locally that test passes, there's something funky with the CI

shanehughes1990 committed:
Making linter happy.

shanehughes1990 committed:
fix(ci): replace deprecated linters in golangci-lint config

- Replace exportloopref with copyloopvar (deprecated since v1.60.2)
- Replace megacheck with gosimple, staticcheck, unused (deprecated)
- Remove deprecated config fields: enable-all, disable-all, presets, fast
- Fixes golangci-lint exit code 7 error in GitHub Actions

shanehughes1990 commented:
@vansante turns out your lint settings are depreciated I have this passing now in an act runner both workflows 👌

vansante commented:
So it looks like ffprobe is crashing on the HDR file, Ill have a look

shanehughes1990 commented:
Lol I was going to upgrade it to V2 too wasn't sure if you would approve of that 😆.

vansante commented:
Yeah, I like to upgrade to latest, that way you don't run into problems again soon (hopefully!) :)

Now, lets see why this test is failing.

vansante commented:
So my guess is that the CI environment has a very restrained memory limit, and decoding a 4K HDR file might be more than it can handle, so I guess we'll just have to take the result json locally, and test with that instead of the file.

shanehughes1990 commented:
I'm working atm, let me give it a go see if I can finagle it, it runs on act so I'm not sure what's up with githubs workers you may be right. If I can't figure it out Ill get copilot to dump the file spec to json and replace the tests.

shanehughes1990 committed:
Merge upstream modernize-ci changes - upgrade golangci-lint config to v2

shanehughes1990 committed:
Refactor GitHub Actions workflow for improved readability and FFmpeg installation

shanehughes1990 committed:
Add detailed debug information for runner environment in GitHub Actions workflow

shanehughes1990 committed:
Refactor FFmpeg installation in GitHub Actions and add HDR JSON test data

shanehughes1990 commented:
Alrighty yeah no idea.... I converted the tests to read the json file I left the .ts there incase we ever find a fix 🤷 , https://github.com/shanehughes1990/go-ffprobe/actions/runs/18364127530/job/52313436298. I enabled workflows on mine so I can test not using act locally (as that passes btw), proof there it finally passes.

Just squash my shameful commit testing 🫤

vansante commented on assets/test.ts.json:
```diff
@@ -0,0 +1,77 @@
+{
```
Is it me, or does this JSON not contain any of the FlexFloat fields?

shanehughes1990 replied:
It's being fixed I didn't realize copilot took it literally. it legit hardcoded hdr json data in the test.

vansante commented on sidedata_test.go:
```diff
@@ -0,0 +1,1391 @@
+}
+
+// Test_ProbeAllAssets tests that all asset files can be probed without error
+func Test_ProbeAllAssets(t *testing.T) {
```
I guess we can stretch the funlen limit a bit, or, we can just add a 
```
//nolint: funlen
```
comment for these functions (as long as they are testing functions)

shanehughes1990 commented:
Oh I'm currently getting copilot to make then adhere moving most of the test stuff to helper test functions standby just a minute.

shanehughes1990 committed:
Enhance HDR metadata handling in tests and update golangci-lint workflow for consistency

vansante commented:
I wanted to remove some unnecessary changes (the github actions formatting and the tests that only get skipped), so I merged it in a separate PR.
